addons/hr_timesheet/controllers/project.py

```diff
@@ -50,4 +50,9 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
         values['allow_timesheets'] = task.allow_timesheets
         values['timesheets'] = timesheets
         values['is_uom_day'] = request.env['account.analytic.line']._is_timesheet_encode_uom_day()
+        parent_task = values['task']
+        sub_task_ids = parent_task._get_subtask_ids_per_task_id()[parent_task.id]
+        sub_task_ids.append(parent_task.id)
+        search_domain = expression.AND([domain, [('parent_task_id', 'in', sub_task_ids)]])
+        values['enable_sub_task_link'] = len(request.env['account.analytic.line'].sudo().search(search_domain, limit=1))
```

Comment:
is it not possible to do a search_count here ? it should be better for performance as it doesn't order the results before returning them